### ISSUE TITLE
ES2015 modules can access `module` and `exports`

### ISSUE THREAD

**Issue by Kovensky** [webpack/webpack#3491]
With the `2.2.0-rc.0` release, any module that has an `include` declaration is also considered a harmony module and does not go through the commonjs module interop.

However, such modules are still allowed to modify the `module` object's `exports`, so they can still make commonjs exports.

This is a breaking change when a file has an `import` but exports using `module.exports`, and is `import`ed via a default import declaration. The `module.exports` object is treated as if it was a namespace, and accessing `default` (most likely) results in `undefined`.

On real ES2015 modules, `module` or `exports` will no longer exist, so these variables should not be accessible (accessing should throw `ReferenceError` unless the user declares them in their own scope).

If you're going with this breaking change, I'd say you should go all the way 😄 

```js
// a.js
import './empty'

module.exports = { foo () {} } // `module` should be ReferenceError instead
```

```js
// empty.js
```

```js
// b.js
import a from './a'
a.foo() // TypeError as of 2.2.0-rc.0
```

**Comment by sokra** [comment]
Yep accessing module.exports should be an errror

**Comment by Noiwex** [comment]
Not quite getting what you've tried to say with b.js example

**Comment by jouni-kantola** [comment]
I'm guessing this is exactly what I'm getting here: https://github.com/webpack/webpack/issues/3719

@sokra: By your answer I take it as we should migrate code meanwhile/before moving over to Webpack 2. I don't know how common the use case is, that `import`/`module.exports` are mixed.

**Comment by TheLarkInn** [comment]
@sokra this is fixed now with #3580 right?

**Comment by mjesun** [comment]
@sokra Is there any option to disable this behavior?
